packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -45,6 +45,11 @@ type DependencyData<TSplitCondition> = $ReadOnly<{
   // If left unspecified, then the dependency is unconditionally split.
   splitCondition?: TSplitCondition,
   locs: Array<BabelSourceLocation>,
+
```

Comment:
We need a way to distinguish `require.context` from all other dependency types, so we know not to try to resolve its name as a file. The problem is this data structure is a bit of a mess - it should probably be a disjoint union with a `type` discriminant.

To avoid a painful refactor (this early in the process) we can add something like `contextParams: $ReadOnly<{recursive: boolean, filter: string}> | null`, along the same lines as `asyncType` and `splitCondition`, such that the dependency is `require.context` iff `contextParams` is nonnull.

Reply:
By the way, looks like Webpack's `require.context` supports a `mode` parameter - let's add at least basic support for that too. (`sync`, `lazy`, and `lazy-once` seem pretty straightforward and valuable; we *might* also be able to implement `weak` but I'm less sure about that one.) https://webpack.js.org/api/module-methods/#requirecontext